query-planner-js/src/composedSchema/buildComposedSchema.ts

```diff
@@ -226,15 +256,11 @@ directive without an @${ownerDirective.name} directive`,
 
 type NamedSchemaElement = GraphQLDirective | GraphQLNamedType;
 
-function parseFieldSet(source: string): FieldSet {
-  const selections = parseSelections(source);
-
+// This should never happen, hence 'programming error', but this assertion
+// guarantees the existence of `graph`.
+function assertGraphFound(graph: Graph | undefined, graphName: string, directiveName: string): asserts graph {
   assert(
-    selections.every(isASTKind('Field', 'InlineFragment')),
-    `Field sets may not contain fragment spreads, but found: "${source}"`,
+    graph,
+    `Programming error: found unexpected \`graph\` argument value "${graphName}" in @${directiveName} directive`,
```

Comment:
I don't necessarily object to "Programming error" but I think these can happen if the definition of the directive in the schema is incorrect — the reason we assume it has to work is that we think the argument's type is the enum and all enum values are in the map, but if the argument is defined as taking, say, a string then you can hit this error, right?  (I think it's probably fine to have this error message, and we can consider validating directive definitions later.)

Reply:
Opened an issue for this: https://github.com/apollographql/federation/issues/678

Good catch in imagining the possible edge case here!